src/ir_passes/mono/monodeferred.rs

```diff
@@ -161,34 +186,26 @@ impl<'a, 'pass: 'a> MonoDeferred<'a, 'pass> {
 
         let port = self
             .monosig
-            .port(self.underlying, self.pass, Underlying::new(*port))
-            .idx();
+            .port_use(self.underlying, Underlying::new(*port))
+            .get();
 
         // generate end expression
-        let end = self.monosig.expr(
-            self.underlying,
-            self.pass,
-            Underlying::new(*end),
-        );
+        let end = self.monosig.expr(self.underlying, Underlying::new(*end));
 
         // convert to concrete value
         let end = self
             .monosig
             .base
-            .bin(self.monosig.base.get(end.idx()).clone());
+            .bin(self.monosig.base.get(end.get()).clone());
```

Comment:
Should probably rename `bin` to something else, since I wrote it for simplifying binop expressions but now it's used for simplifying everything. I'm also a little out of the loop with some of the simplify stuff that you and/or Edmund wrote, so maybe there is some better machinery for doing this